packages/mermaid/src/diagram-api/diagram-orchestration.ts

```diff
@@ -55,25 +55,28 @@ export const addDiagrams = () => {
       return text.toLowerCase().trimStart().startsWith('---');
     }
   );
+  // Ordering of detectors is important. The first one to return true will be used.
   registerLazyLoadedDiagrams(
     error,
     c4,
-    classDiagram,
     classDiagramV2,
+    classDiagram,
     er,
     gantt,
     info,
     pie,
     requirement,
     sequence,
+    flowchartElk,
+    // TODO @knsv: Should v2 come before flowchart?
```

Comment:
This is fine. The legacy detector only triggers on graph and if the default renderer has not been configured to be dagre-wrapper this one, (flowchart), should kick in.